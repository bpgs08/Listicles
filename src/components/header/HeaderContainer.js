import React from "react";
import { connect } from "react-redux";
import Header from "./Header";

const HeaderContainer = ({ searchParam, currentPage }) => {
  return <Header searchParam={searchParam} currentPage={currentPage} />;
};

const mapStateToProps = ({ addArticlesReducer }) => {
  return {
    searchParam: addArticlesReducer.searchParam,
    currentPage: addArticlesReducer.currentPage,
  };
};

export default connect(mapStateToProps)(HeaderContainer);
