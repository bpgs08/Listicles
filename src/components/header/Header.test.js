import React from "react";
import Enzyme, { mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import { Provider } from "react-redux";
import store from "../../store";
import renderer from "react-test-renderer";
import HeaderContainer from "./HeaderContainer";
Enzyme.configure({ adapter: new Adapter() });

describe("<HeaderContainer /> component", () => {
  it("snapshot", () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <HeaderContainer />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should find app title correctly", () => {
    const wrapper = mount(
      <Provider store={store}>
        <HeaderContainer searchParamLength={null} />
      </Provider>
    );
    expect(wrapper.find("h1.App-title").text()).toEqual("Top Headlines");
  });
});
