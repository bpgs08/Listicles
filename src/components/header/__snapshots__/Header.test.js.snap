// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<HeaderContainer /> component snapshot 1`] = `
<div
  className="sc-AxjAm kqqGFW App-header"
>
  <h1
    className="App-title"
  >
    Top Headlines
  </h1>
  <h2 />
</div>
`;
